includes/Handlers/Connection.php

```diff
@@ -301,7 +301,10 @@ public function handle_connect() {
 			$this->update_system_user_id( $system_user_id );
 			$this->update_installation_data();
 
-			facebook_for_woocommerce()->get_products_sync_handler()->create_or_update_all_products();
+			// Only trigger initial full batch-api sync if the store doesn't have large product count.
+			if ( $facebook_for_woocommerce()->get_integration()->allow_full_batch_api_sync() ) {
+				facebook_for_woocommerce()->get_products_sync_handler()->create_or_update_all_products();
+			}
```

Comment:
We can define `wc_facebook_enable_product_sync` as `false` here to avoid BG sync running in the background
```suggestion
			// Only trigger initial full batch-api sync if the store doesn't have large product count.
			if ( $facebook_for_woocommerce()->get_integration()->allow_full_batch_api_sync() ) {
				facebook_for_woocommerce()->get_products_sync_handler()->create_or_update_all_products();
			} else  {
				update_option( \WC_Facebookcommerce_Integration::SETTING_ENABLE_PRODUCT_SYNC, 'no' );
			}
```

Reply:
Is this needed to prevent a full sync? I don't think we want to completely disable product sync, just prevent a full catalog sync using `Background` job class (which has issue with the huge `wp_option`). 

If we do need to disable the feature like this, we run the risk of merchant re-enabling it in the UI.